Blocks/Persistence/file.h

```diff
@@ -133,7 +134,7 @@ class FilePersister {
     std::fstream head_rewriter;
 
     // `offset.size() == end.next_index`, and `offset[i]` is the offset in bytes where the line for index `i` begins.
-    std::mutex mutex;
+    std::mutex& mutex;  // Guards `offset`, `head_offset` and `timestamp`.
```

Comment:
Let's rename `mutex` to reflect that it's a reference to an external mutex object. I'd suggest something like `mutex_ref` or `external_mutex_ref`.